auth/tests/api/helpers.rs

```diff
@@ -137,6 +135,20 @@ impl MockedStripeServer {
     async fn subscription_retrieve_handler(
         Path(subscription_id): Path<String>,
     ) -> axum::response::Response<String> {
+        static TOGGLE: Mutex<bool> = Mutex::new(false);
```

Comment:
We can avoid using statics and store this toggle on the `MockedStripeServer`.
Also, naming it as `subscription_cancel_side_effect_toggle` would be great.
What do you think?